Public/Src/Tools/SymbolDaemon/SymbolFile.cs

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using BuildXL.Cache.ContentStore.Hashing;
+using BuildXL.Ipc.ExternalApi;
+using BuildXL.Utilities;
+using Microsoft.VisualStudio.Services.Symbol.Common;
+using Tool.ServicePipDaemon;
+
+namespace Tool.SymbolDaemon
+{
+    /// <summary>
+    /// File and its symbol data.
+    /// </summary>
+    public sealed class SymbolFile
+    {
+        /// <summary>
+        /// Prefix for the error message of the exception that gets thrown when a materialized file is a symlink.
+        /// </summary>
+        private const string MaterializationResultIsSymlinkErrorPrefix = "File materialization succeeded, but file found on disk is a symlink: ";

```

Comment:
Why do we fail when the file is a symlink? We're not uploading the file, are we? We're uploading the symbols.

Reply:
If the endpoint has the content referenced in the DebugEntry, indeed, we are not going to upload the file again. However, if the content is not there, the endpoint will refuse the DubugEntry and we'll need to upload the file.

---
In reply to: [313640172](https://github.com/microsoft/BuildXL/pull/727#discussion_r313640172) [](ancestors = 313640172)